Performance tests: some more polishing

tamiko opened the pull request:
This pull request contains a number of small fixes for the testsuite: 

 - Performance tests: limit concurrency of tests
 - Performance tests: run serial timing tests exclusively
 - Documentatoin updates
 - CMake: add an .exclusive. toggle to ensure that certain tests always run
   in serial

luca-heltai reviewed (approved):
Minor typo. Nice!

kronbichler reviewed (approved):
Don't you need the exclusive keyword also for the step-37 performance test? Or is that covered by MPI?

tamiko commented:
@kronbichler This should already be covered by the `mpirun=max` annotation (as long as one doesn't overschedule with `ctest -j` beyond `TEST_MPI_RANK_LIMIT`). On second thought I have modified the PR and added an exclusive keyword as well.

tamiko commented:
/rebuild
